app/certificates.go

```diff
@@ -0,0 +1,442 @@
+		Bytes: privBytes,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	return certPEMBuffer.Bytes(), certPrivateKeyPEMBuffer.Bytes(), nil
+}
+
+func NewCertificatesCommand() (CommandOut, error) {
+	return CommandOut{
+		Command: &cli.Command{
+			Name:    "generate-certificates",
+			Aliases: []string{"gen"},
+			Usage:   "Generate tls certificates",
+			Subcommands: []*cli.Command{
+				{
+					Name:    "certificate-authority",
+					Usage:   "Generate a certificate authority",
+					Aliases: []string{"ca"},
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     "organization",
+							Usage:    "The name of the organization",
+							Aliases:  []string{"org"},
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     "duration",
+							Usage:    "The duration of the ca certificate, for example: 1y30d (1 year and 30 days)",
+							Aliases:  []string{"d"},
+							Required: true,
+							Action: func(_ *cli.Context, v string) error {
+								d, err := utils.ParseDuration(v)
+								if err != nil {
+									return fmt.Errorf("failed to parse duration: %v", err)
+								}
+								if d > maxCADuration {
+									return fmt.Errorf("duration cannot be more than: %s", maxCADuration)
+								}
+								if d <= minCADuration {
+									return fmt.Errorf("duration cannot be less than: %s", minCADuration)
+								}
+								return nil
+							},
+						},
+						&cli.PathFlag{
+							Name:     CaCertificateFileFlagName,
+							Usage:    "The path of the file to store the generated certificate-authority in",
+							Aliases:  []string{"ca-cert"},
+							Required: true,
+						},
+						&cli.PathFlag{
+							Name:     CaPrivateKeyFileFlagName,
+							Usage:    "The path of the file to store the generated certificate-authority's private key in",
+							Aliases:  []string{"ca-key"},
+							Required: true,
+						},
+						&cli.BoolFlag{
+							Name:    "rsa-algorithm",
+							Aliases: []string{"rsa"},
+							Usage:   "Generate the certificate-authority using the RSA algorithm instead of ecdsa (optional)",
+						},
+					},
+					Action: func(ctx *cli.Context) error {
+						duration, err := utils.ParseDuration(ctx.String("duration"))
+						if err != nil {
+							return err
+						}
+						caPem, caPrivKey, err := generateCACertificate(generateCACertificateInput{
+							Organization: ctx.String("organization"),
+							Duration:     duration,
+							RSAAlgorithm: ctx.Bool("rsa-algorithm"),
+						})
+						if err != nil {
+							return fmt.Errorf("failed to generate ca certificate: %s", err)
+						}
+						yes, err := ConfirmPrompt(
+							ctx,
+							fmt.Sprintf("storing certificate authority's private key at %s, do not share this key with anyone. confirm: ",
+								ctx.Path(CaPrivateKeyFileFlagName),
+							),
+						)
+						if err != nil || !yes {
+							return nil
+						}
+						err = ioutil.WriteFile(
+							ctx.Path(CaCertificateFileFlagName),
+							caPem,
+							0600,
```

Comment:
[Nit] does CA certificate need same permissions as the private key? Being able to pass around the public key I think is fair game and this makes it a bit harder to do if we set 0600 for the public key as well as the private?